clients/apple/Shared/Views/AppView.swift

```diff
@@ -17,6 +17,32 @@ struct AppView: View {
                 switch files.root {
                 case .some(let root):
                     BookView(currentFolder: root, account: account)
+                        .onOpenURL() { url in
+                            guard let uuidString = url.host, let id = UUID(uuidString: uuidString), url.scheme == "lb" else {
+                                DI.errors.errorWithTitle("Malformed link", "Cannot open file")
+                                return
+                            }
+        
+                            DispatchQueue.global(qos: .userInitiated).async {
+                                while !DI.files.hasRootLoaded {
+                                    if DI.accounts.calculated && DI.accounts.account == nil {
+                                        return
+                                    }
+                                }
```

Comment:
my gut says this is a very aggressive way to express this idea, and is not friendly to the scheduler. I don't know the details of how a lot of this plays out in swift / swiftui, but generally adding a sleep(1) at the moment of loop iteration is enough of a hacky signal to the scheduler to try to do something else meanwhile.

Where this performs poorly is at startup the loop iterates like 10k times in 100ms making no progress while the scheduler doesn't understand that it's making no progress. This will manifest itself as startup lag.

Reply:
This isn't new to this PR, it's somewhat always how I have handled opening links within Lockbook. I agree with your sentiment, I can fix it in a future PR.

Reply:
just add a sleep(1) for my soul